AppleAuthSampleProject/Assets/AppleAuthSample/Editor/SignInWithApplePostprocessor.cs

```diff
@@ -1,44 +1,46 @@
-#if UNITY_IOS || UNITY_TVOS
+#if UNITY_IOS || UNITY_TVOS || UNITY_VISIONOS
 #define UNITY_XCODE_EXTENSIONS_AVAILABLE
 #endif
 
 using AppleAuth.Editor;
 using UnityEditor;
 using UnityEditor.Callbacks;
+using UnityEngine;
 #if UNITY_XCODE_EXTENSIONS_AVAILABLE
 using UnityEditor.iOS.Xcode;
 #endif
 
-namespace AppleAuthSample.Editor
-{
-    public static class SignInWithApplePostprocessor
-    {
-        private const int CallOrder = 1;
+namespace AppleAuthSample.Editor {
+  public static class SignInWithApplePostprocessor {
+    private const int CallOrder = 1;
 
-        [PostProcessBuild(CallOrder)]
-        public static void OnPostProcessBuild(BuildTarget target, string path)
-        {
-            if (target == BuildTarget.iOS || target == BuildTarget.tvOS)
-            {
-                #if UNITY_XCODE_EXTENSIONS_AVAILABLE
-                    var projectPath = PBXProject.GetPBXProjectPath(path);
-                    #if UNITY_2019_3_OR_NEWER
-                        var project = new PBXProject();
-                        project.ReadFromString(System.IO.File.ReadAllText(projectPath));
-                        var manager = new ProjectCapabilityManager(projectPath, "Entitlements.entitlements", null, project.GetUnityMainTargetGuid());
-                        manager.AddSignInWithAppleWithCompatibility(project.GetUnityFrameworkTargetGuid());
-                        manager.WriteToFile();
-                    #else
+    [PostProcessBuild(CallOrder)]
+    public static void OnPostProcessBuild(BuildTarget target, string path) {
+      if (target == BuildTarget.iOS || target == BuildTarget.tvOS || target == BuildTarget.VisionOS) {
+#if UNITY_XCODE_EXTENSIONS_AVAILABLE
+        string projectPath = PBXProject.GetPBXProjectPath(path);
+        if (target == BuildTarget.VisionOS) {
+            
+          // This is a temporary fix for the Unity Editor's bug:
+          // After switch to VisionOS platform the projectPath is still "xx/Unity-iPhone.xcodeproj/project.pbxproj",
+          // while the expected path is "xx/Unity-VisionOS.xcodeproj/project.pbxproj",
+          projectPath = projectPath.Replace("Unity-iPhone.xcodeproj", "Unity-VisionOS.xcodeproj");
+        }
+#if UNITY_2019_3_OR_NEWER
+        var project = new PBXProject();
+        project.ReadFromString(System.IO.File.ReadAllText(projectPath));
+        var manager = new ProjectCapabilityManager(projectPath, "Entitlements.entitlements", null, project.GetUnityMainTargetGuid());
+        manager.AddSignInWithAppleWithCompatibility(project.GetUnityFrameworkTargetGuid());
+        manager.WriteToFile();
+#else
```

Comment:
Most of these changes are not justified. Just add the required check for VisionOS. in the current code.
Also, we need to protect the VisionOS definition, since it's does not exist in other Unity versions.
From what I can see, VisionOS module was introduced in Unity 2022.3.5f1, so let's put 2022.3 as the basis

```
#if UNITY_2022_3_OR_LATER
    if (target == BuildTarget.VisionOS)
    {
        projectPath = projectPath.Replace("Unity-iPhone.xcodeproj", "Unity-VisionOS.xcodeproj");
    }
#endif
```

Reply:
done,

Reply:
To achieve this, i separate the VisionOS part with the rest

Reply:
verified by build and run in simulator